addons/hr_recruitment/migrations/9.0.1.0/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,113 @@
+---Fields in module 'hr_recruitment'---
+hr_recruitment / hr.applicant             / attachment_ids (one2many)     : NEW relation: ir.attachment
+# Nothing to do
+
+hr_recruitment / hr.applicant             / availability (integer)        : type is now 'date' ('integer')
+# Unable to calculate date form a integer --> Nothing to do
+
+hr_recruitment / hr.applicant             / campaign_id (many2one)        : NEW relation: utm.campaign
+# Nothing to do
+
+hr_recruitment / hr.applicant             / categ_ids (many2many)         : relation is now 'hr.applicant.category' ('hr.applicant_category')
+# Nothing to do
+
+hr_recruitment / hr.applicant             / medium_id (many2one)          : NEW relation: utm.medium
+# Nothing to do
+
+hr_recruitment / hr.applicant             / message_follower_ids (many2many): not a function anymore
+hr_recruitment / hr.applicant             / message_follower_ids (many2many): relation is now 'mail.followers' ('res.partner')
+hr_recruitment / hr.applicant             / message_follower_ids (many2many): type is now 'one2many' ('many2many')
+# Concerns module Mail (has to be done in another migration script specifically for that module) ---> Nothing to do
+
+hr_recruitment / hr.applicant             / priority (selection)          : selection_keys is now '['0', '1', '2', '3']' ('['0', '1', '2', '3', '4']')
+# Done: Adapt to new range
+
+hr_recruitment / hr.applicant             / source_id (many2one)          : relation is now 'utm.source' ('hr.recruitment.source')
+# Nothing to do
```

Comment:
You need to create an utm.source for each of the existing hr.recruitment.source, as now the object inherits (via delegation) of this object, and fill the corresponding field:
````
cr.execute('INSERT INTO utm_source ... RETURNING id')
new_id = cr.fetchone()
cr.execute('UPDATE hr_applicant SET source_id = %s WHERE id = %s', (new_id, old_id))
```